src/pss/signing_key.rs

```diff
@@ -90,69 +85,46 @@ where
 // `*Signer` trait impls
 //
 
-impl<D> RandomizedDigestSigner<D, Signature> for SigningKey<D>
+impl<D> RandomizedDigestSigner<D, Signature<D>> for SigningKey<D>
 where
     D: Digest + FixedOutputReset,
 {
     fn try_sign_digest_with_rng<R: TryCryptoRng + ?Sized>(
         &self,
         rng: &mut R,
         digest: D,
-    ) -> signature::Result<Signature> {
+    ) -> signature::Result<Signature<D>> {
         sign_digest::<_, D>(rng, false, &self.inner, &digest.finalize(), self.salt_len)?
             .as_slice()
             .try_into()
     }
 }
 
-impl<D> RandomizedSigner<Signature> for SigningKey<D>
-where
-    D: Digest + FixedOutputReset,
-{
-    fn try_sign_with_rng<R: TryCryptoRng + ?Sized>(
-        &self,
-        rng: &mut R,
-        msg: &[u8],
-    ) -> signature::Result<Signature> {
-        self.try_sign_digest_with_rng(rng, D::new_with_prefix(msg))
-    }
-}
-
-impl<D> RandomizedPrehashSigner<Signature> for SigningKey<D>
+impl<D> RandomizedPrehashSigner<Signature<D>> for SigningKey<D>
 where
     D: Digest + FixedOutputReset,
 {
     fn sign_prehash_with_rng<R: TryCryptoRng + ?Sized>(
         &self,
         rng: &mut R,
         prehash: &[u8],
-    ) -> signature::Result<Signature> {
+    ) -> signature::Result<Signature<D>> {
         sign_digest::<_, D>(rng, false, &self.inner, prehash, self.salt_len)?
             .as_slice()
             .try_into()
     }
 }
 
 #[cfg(feature = "os_rng")]
-impl<D> PrehashSigner<Signature> for SigningKey<D>
+impl<D> PrehashSigner<Signature<D>> for SigningKey<D>
 where
     D: Digest + FixedOutputReset,
 {
-    fn sign_prehash(&self, prehash: &[u8]) -> signature::Result<Signature> {
+    fn sign_prehash(&self, prehash: &[u8]) -> signature::Result<Signature<D>> {
         self.sign_prehash_with_rng(&mut OsRng, prehash)
     }
 }
 
-#[cfg(feature = "os_rng")]
-impl<D> Signer<Signature> for SigningKey<D>
-where
-    D: Digest + FixedOutputReset,
-{
-    fn try_sign(&self, msg: &[u8]) -> signature::Result<Signature> {
-        self.try_sign_with_rng(&mut OsRng, msg)
-    }
-}
```

Comment:
I guess the alternative would be to preserve these `Signer` impls and not impl `PrehashSignature` for `Signature` since it's not actually known at the type level

Reply:
Iirc I tried that, but I came in conflict with the blanket implementation.

can’t try it right this moment, will try that in an hour or so.

Reply:
If you don't `impl PrehashSignature` the blanket impl shouldn't matter